File: src/main/java/com/server/ordering/domain/member/MemberBase.java
package com.server.ordering.domain.member;

import com.server.ordering.domain.PhoneNumber;
import lombok.*;

import javax.persistence.*;

import static javax.persistence.CascadeType.ALL;
import static javax.persistence.FetchType.*;
import static lombok.AccessLevel.*;

@Getter
@MappedSuperclass
@RequiredArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public abstract class MemberBase {

    @NonNull
    protected String signInId;
    @NonNull
    protected String password;

    @NonNull
    @OneToOne(fetch = LAZY, cascade = ALL, orphanRemoval = true)
    @JoinColumn(name = "phone_number")
    protected PhoneNumber phoneNumber;

    public void putPassword(String password) {
        this.password = password;
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/member/MemberBase.kt
package com.byeoru.ordering_server.domain.member

import com.byeoru.ordering_server.domain.PhoneNumber
import javax.persistence.*

@MappedSuperclass
abstract class MemberBase(signInId: String,
                          password: String,
                          phoneNumber: PhoneNumber) {

    var signInId: String = signInId
        protected set
    var password: String = password
        protected set

    @OneToOne(fetch = FetchType.LAZY, cascade = [CascadeType.ALL], orphanRemoval = true)
    @JoinColumn(name = "phone_number")
    var phoneNumber: PhoneNumber = phoneNumber
        protected set

    var firebaseToken: String? = null
        protected set

    fun putPassword(password: String) {
        this.password = password
    }

    fun putFirebaseToken(firebaseToken: String?) {
        this.firebaseToken = firebaseToken
    }

    fun clearFirebaseToken() {
        firebaseToken = null
    }
}